### PR TITLE
DHFPROD-3992: Add mapping step to flow and run it 

### PR DESCRIPTION
**Pull request by srinathgit**
### Description
This PR contains the following
1. Front end code for adding/running a mapping step in a  flow
2. Backend code and tests for adding/running a mapping step in a flow

@wooldridge is working on front end tests and will add them in new PR. Since the backend work is required for other stories, I am creating the PR now

#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Comment by bsrikan** [comment]
@srinathgit I unchecked Squash, since this PR needs to squash the commits

File: marklogic-data-hub-central/ui/src/pages/EntityTypes.tsx [hunk 1]
```diff
@@ -43,18 +47,92 @@ const EntityTypes: React.FC = () => {
...
 
     }
+
+  //GET all the flow artifacts
+  const getFlows = async () => {
+        try {
+            let response = await axios.get('/api/flows');
+            if (response.status === 200) {
+                setFlows(response.data);
+                console.log('GET flows successful', response);
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/pages/EntityTypes.tsx]: Can we please get rid of all the console.log and only have console.error where applicable.

File: marklogic-data-hub-central/ui/src/pages/EntityTypes.tsx [hunk 2]
```diff
@@ -43,18 +47,92 @@ const EntityTypes: React.FC = () => {
...
+        } catch (error) {
+            let message = error.response.data.message;
+            console.log('Error getting flows', message);
+        } finally {
+        resetSessionTime();
+        }
+    }
+
+  // POST load data step to new flow
+  const addStepToNew = async () => {
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/pages/EntityTypes.tsx:64]: Are these specific to loadData?

File: marklogic-data-hub-central/ui/src/pages/EntityTypes.tsx [hunk 3]
```diff
@@ -43,18 +47,92 @@ const EntityTypes: React.FC = () => {
...
+        console.log('Error while adding load data step to new flow.', message);
+        setIsLoading(false);
+        handleError(error);
+    } finally {
+      resetSessionTime();
+    }
+  }
+
+  // POST load data step to existing flow
+  const addStepToFlow = async (loadArtifactName, flowName) => {
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/pages/EntityTypes.tsx]: Same as above. Is this specific to loadData?

File: marklogic-data-hub-central/ui/src/pages/EntityTypes.tsx [hunk 4]
```diff
@@ -43,18 +47,92 @@ const EntityTypes: React.FC = () => {
...
+        "loadData": { 
+          "name": loadArtifactName
+        }
+      }
+    };
+    try {
+      setIsLoading(true);
+      let url = '/api/flows/' + flowName + '/steps';
+      let body = stepToAdd;
+      let response = await axios.post(url, body);
```
Comment (bsrikan) [inline marklogic-data-hub-central/ui/src/pages/EntityTypes.tsx:99]: We should try and refactor api calls under " marklogic-data-hub-central/ui/src/api/" for anything that gets committed to develop. I will create a story for refactoring existing axios get and post requests in develop branch.

Comment (srinathgit) [reply marklogic-data-hub-central/ui/src/pages/EntityTypes.tsx:98]: Refactoring will be done in separate story

**Comment by srinathgit** [comment]
> @srinathgit I unchecked Squash, since this PR needs to squash the commits

I have 2 commits, one for frontend and backend. I spoke to @SameeraPriyathamTadikonda  and he is good with it

File: marklogic-data-hub-central/ui/src/pages/EntityTypes.tsx (hunk 4 above)
Comment (bsrikan) [reply marklogic-data-hub-central/ui/src/pages/EntityTypes.tsx:98]: Refactoring existing code can be part of a separate story. These are new API requests right?

Comment (srinathgit) [reply marklogic-data-hub-central/ui/src/pages/EntityTypes.tsx:98]: Adding loadData, mapping and other artifacts to flow will all be using the same endpoint which will be created in that new refactoring storing .

**Comment by srinathgit** [comment]
The PR just requires "Reviewed Tests" to be checked to get merged

**Pull request merged**